### PR TITLE
Update migration test for TEE integration

### PR DESCRIPTION
**Pull request by zacshowa**
This commit updates the migration-test.bash script to reflect the new migration flow for the TEE integration. It also adds some utility scripts and updates some exsisting example files to assist in the migration test. This commit is the result of creating a new branch off of celestia-integration, and checking out the individual files from my work on the espresso-tee-migration branch, and then making a couple edits to address feedback in the previous pr. 

Closes #[333](https://github.com/EspressoSystems/nitro-espresso-integration/issues/333)

### This PR:
Updates the migration-test.bash script, and associated utilities to enable testing a live migration to the Espresso TEE integration. 

### This PR does not:
Update any README files with the new migration flow, that will be a separate PR in the orbit-actions repo.

### Key places to review:
Changes to migration-test.bash.

### How to test this PR: 
from the nitro-testnode repo, run the script located at `espresso-tests/migration-test.bash` if the script finishes, the migration test was successful. Note this test takes upwards of 10 minutes to run most of the time.

### Things tested:
That the migration-test.bash script is able to be successfully run, and that the resulting network configuration is able to accept transactions, and post batches by using the sendl2 command from the scripts docker container.

### CODE REVIEW

File: scripts/index.ts [hunk 1]
```diff
@@ -41,6 +41,8 @@ async function main() {
       espressoUrl: { string: true, description: 'Espresso Sequencer url', default: 'http://espresso-dev-node:41000' },
       lightClientAddress: { string: true, description: 'address of the light client contract', default: ''},
       enableEspressoFinalityNode: {boolean: true, description: 'enable finality node', default: false},
+      simpleWithValidator: {boolean: true, description: 'start a simple node that validates', default: false},
+      migration: {boolean :true, description: 'write configs compatible with the migration test', default: false},
```
Comment (ImJeremyHe) [inline scripts/index.ts]: I believe this `migration` flag should be unnecessary

File: espresso-tests/migration-test.bash [hunk 2]
```diff
@@ -179,4 +183,3 @@ echo "Confirmed nodes have progressed"
 # Echo to signal that test has been successful
 echo "Migration successfully completed!"
 
-docker compose down
```
Comment (ImJeremyHe) [inline espresso-tests/migration-test.bash]: this is an unnecessary change

File: espresso-tests/test-chain-config.json
Comment (ImJeremyHe) [inline espresso-tests/test-chain-config.json:1]: Remove it?

File: docker-compose.yaml [hunk 3]
```diff
@@ -168,6 +168,8 @@ services:
       - "config:/config"
       - "tokenbridge-data:/tokenbridge-data"
     command: --conf.file /config/sequencer_config.json --node.feed.output.enable --node.feed.output.port 9642  --http.api net,web3,eth,txpool,debug --node.seq-coordinator.my-url  ws://sequencer:8548 --graphql.enable --graphql.vhosts * --graphql.corsdomain *
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (Sneh1999) [inline docker-compose.yaml]: do we need this change?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ on:
 
 jobs:
   build_and_run:
-    runs-on: ubuntu-8
+    runs-on: ubuntu-latest
```
Comment (alysiahuggins) [inline .github/workflows/ci.yml:15]: I'm not familiar with the TEE set up but could this break the environment with unexpected changes in OS versions?

File: espresso-tests/migration-test.bash [hunk 5]
```diff
@@ -50,51 +51,64 @@ CHILD_CHAIN_UPGRADE_EXECUTOR_ADDRESS=$(cast call $L1_TOKEN_BRIDGE_CREATOR_ADDRES
...
-echo "Deploying Espresso Osp"
-# Change directory to orbit actions dir for the following commands.
 cd $ORBIT_ACTIONS_DIR
-# ** Essential migration step ** Forge script to deploy new OSP entry. We do this to later point the rollups challenge manager to the espresso integrated OSP.
-forge script --chain $PARENT_CHAIN_CHAIN_ID contracts/parent-chain/espresso-migration/DeployEspressoOsp.s.sol:DeployEspressoOsp --rpc-url $PARENT_CHAIN_RPC_URL --broadcast -vvvv
+forge update
+echo "Deploying mock espresso tee verifier"
+forge script --chain $PARENT_CHAIN_CHAIN_ID ../espresso-tests/DeployMockVerifier.s.sol:DeployMockVerifier --rpc-url $PARENT_CHAIN_RPC_URL --broadcast -vvvv
+
+ESPRESSO_TEE_VERIFIER_ADDRESS=$(cat broadcast/DeployMockVerifier.s.sol/1337/run-latest.json | jq -r '.transactions[0].contractAddress' | cast to-checksum)
```
Comment (alysiahuggins) [inline espresso-tests/migration-test.bash:59]: the use of chain ID 1337 suggests this setup is primarily for local testing. Do we ever test with a fork and thus need the chain id to change?

File: espresso-tests/migration-test.bash [hunk 6]
```diff
@@ -50,51 +51,64 @@ CHILD_CHAIN_UPGRADE_EXECUTOR_ADDRESS=$(cast call $L1_TOKEN_BRIDGE_CREATOR_ADDRES
...
+echo "Deploying ArbOS Upgrade action"
+# Forge script to deploy the Espresso ArbOS upgrade action.
+# ** Essential migration step ** the ArbOS upgrade signifies that the chain is now espresso compatible.
+forge script --chain $CHILD_CHAIN_CHAIN_NAME contracts/child-chain/espresso-migration/DeployArbOSUpgradeAction.s.sol:DeployArbOSUpgradeAction  --rpc-url $CHILD_CHAIN_RPC_URL --broadcast -vvvv
 
-# Get the number of confirmed nodes before the upgrade to ensure the staker is still working.
-NUM_CONFIRMED_NODES_BEFORE_UPGRADE=$(cast call --rpc-url $PARENT_CHAIN_RPC_URL $ROLLUP_ADDRESS 'latestConfirmed()(uint256)')
-# Shutdown nitro node
+# Get the address of the newly deployed upgrade action.
+ARBOS_UPGRADE_ACTION=$(cat broadcast/DeployArbOSUpgradeAction.s.sol/412346/run-latest.json | jq -r '.transactions[0].contractAddress')
```
Comment (alysiahuggins) [inline espresso-tests/migration-test.bash:92]: from a google search this chain id resolves to `Syndr Nitro L3 - Caldera Chain`. I'm not familiar with the set up here but can you share more on this workflow

**Review by alysiahuggins — COMMENTED** [review]
nit: could be nice if there are comments for key services in the docker yaml to explain their use e.g. espresso-dev-node

File: .github/workflows/ci.yml (hunk 4 above)
Comment (zacshowa) [reply .github/workflows/ci.yml:15]: This is just a change to the CI of the nitro-testnode repo. I think this was a change brought in by a previous merge. I'm not even sure if github has ubuntu 8 runners anymore. The CI on this PR passed with the ubuntu 24 runner, so it should be fine, but I can try changing it back.

File: espresso-tests/migration-test.bash (hunk 5 above)
Comment (zacshowa) [reply espresso-tests/migration-test.bash:59]: This is just a local testing thing. As are the other chain ID's. This test wasn't even written to be tested on forks of other chains, but I suppose that could be done in the future if it gives us better testing.

File: espresso-tests/migration-test.bash (hunk 6 above)
Comment (zacshowa) [reply espresso-tests/migration-test.bash:92]: This was also just for local testing.  I didn't even know this was a claimed chain id in the wild.

**Review by Sneh1999 — APPROVED** [review]
LGTM! Awesome work @zacshowa  @ImJeremyHe  🎉 🎉

File: .github/workflows/ci.yml (hunk 4 above)
Comment (alysiahuggins) [reply .github/workflows/ci.yml:15]: ahhh if it's on the nitro CI then we usually just stick with that right? if so, ignore my comment

File: espresso-tests/migration-test.bash (hunk 5 above)
Comment (alysiahuggins) [reply espresso-tests/migration-test.bash:59]: okie, could be a future issue then but sounds low priority

File: espresso-tests/migration-test.bash (hunk 6 above)
Comment (alysiahuggins) [reply espresso-tests/migration-test.bash:92]: hehe , got it

**Review by alysiahuggins — APPROVED** [review]
